frontend/packages/console-app/src/components/cloud-shell/AutoResizeTerminal.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react';
+import { Terminal as XTerminal } from 'xterm';
+import Measure from 'react-measure';
+import * as fit from 'xterm/lib/addons/fit/fit';
+import * as full from 'xterm/lib/addons/fullscreen/fullscreen';
+
+XTerminal.applyAddon(fit);
+XTerminal.applyAddon(full);
+
+type TerminalProps = {
+  onData: any;
+  className?: string;
+};
+
+type TerminalState = {
+  width: number;
+  height: number;
+};
+
+export class Terminal extends React.Component<TerminalProps, TerminalState> {
+  private innerRef;
+
+  private outerRef;
+
+  private onDataReceived;
+
+  private terminal;
+
+  private padding;
+
+  constructor(props) {
+    super(props);
+    this.innerRef = React.createRef();
+    this.outerRef = React.createRef();
+    this.state = {
+      width: 0,
+      height: 0,
+    };
+    this.onDataReceived = (data) => {
+      this.terminal && this.terminal.write(data);
+    };
+    this.padding = 20;
+    const options = {
+      fontFamily: 'monospace',
+      fontSize: 16,
+      cursorBlink: false,
+      cols: 80,
+      rows: 25,
+    };
+    this.terminal = new XTerminal(Object.assign({}, options));
+    this.terminal.on('data', this.props.onData);
+  }
+
+  focus = () => {
+    this.terminal && this.terminal.focus();
+  };
+
+  onResize = () => {
+    const node = this.outerRef.current;
+
+    if (!node) {
+      return;
+    }
+
+    const bodyRect = document.body.getBoundingClientRect();
+    const nodeRect = node.getBoundingClientRect();
+
+    // This assumes we want to fill everything below and to the right.  In full-screen, fill entire viewport
+    // Use body height when node top is too close to pageRect height
+    const { bottom } = bodyRect;
+    const height = Math.floor(bottom - nodeRect.top);
+    const width = Math.floor(bodyRect.width - nodeRect.left);
+
+    if (height === this.state.height && width === this.state.width) {
+      return;
+    }
+    // rerender with correct dimensions
+    this.setState({ height, width }, () => {
+      const { terminal } = this;
+      if (!terminal) {
+        return;
+      }
+      // tell the terminal to resize itself
+      terminal.fit();
+    });
+  };
+
+  onConnectionClosed = (reason) => {
+    const { terminal } = this;
+    if (!terminal) {
+      return;
+    }
+    terminal.write(`\x1b[31m${reason || 'disconnected'}\x1b[m\r\n`);
+    terminal.cursorHidden = true;
+    terminal.setOption('disableStdin', true);
+    terminal.refresh(terminal.y, terminal.y);
+  };
+
+  componentDidMount() {
+    this.terminal.open(this.innerRef.current);
+    this.focus();
+    this.onResize();
+    this.onDataReceived();
```

Comment:
Remove? You're not sending any data.

Reply:
Nothing works without this in place :)

Reply:
You are sending empty data to the terminal though? Why?
How does nothing work without calling `this.terminal.write(undefined)` ?

I don't see this being done in the existing `terminal.jsx`

Reply:
Sorry, my bad..it's just the onmount call, the thought the whole function